src/app/(authenticated)/(dashboard)/[publicId]/settings/profile/page.tsx

```diff
@@ -4,81 +4,237 @@ import Image from "next/image";
 import { useForm } from "react-hook-form";
 import { Input } from "@/components/ui/input";
 import { Button } from "@/components/ui/button";
-import { useSession } from "next-auth/react";
+import { z } from "zod";
+import { Form, FormItem, FormLabel, FormMessage } from "@/components/ui/form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { api } from "@/trpc/react";
+import React, { useEffect, useRef, useState } from "react";
+import { useToast } from "@/components/ui/use-toast";
 
-import {
-  Form,
-  FormControl,
-  FormField,
-  FormItem,
-  FormLabel,
-  FormMessage,
-} from "@/components/ui/form";
+const profileSettingsSchema = z.object({
+  fullName: z.string().min(2).max(40),
+  jobTitle: z.string().min(2).max(30),
+  loginEmail: z.string().email(),
+  workEmail: z.string().email(),
+  avatarUrl: z.string(),
+});
 
 const ProfileSettingsPage = () => {
-  const { data } = useSession();
-  const form = useForm();
+  const { toast } = useToast();
+  const utils = api.useUtils();
+  const memberProfile = api.member.getProfile.useQuery();
 
+  const [file, setFile] = useState<File | null>(null);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  // @ts-expect-error: xxxx
+  const form = useForm<z.infer<typeof profileSettingsSchema>>({
+    resolver: zodResolver(profileSettingsSchema),
+  });
+
+  // @ts-expect-error: xxxx
+  const { errors } = form.formState;
+  const isSubmitting = form.formState.isSubmitting;
+  const isDirty = form.formState.isDirty;
+  const canSave: boolean = file ? isDirty || !!file : isDirty;
+
+  // @ts-expect-error: xxxx
+  const saveProfileMutation = api.member.updateProfile.useMutation({
+    onSuccess: async () => {
+      await utils.member.getProfile.invalidate();
+      return toast({
+        variant: "default",
+        title: "Profile changed successfully.",
+      });
+    },
+    onError: () => {
+      return toast({
+        variant: "destructive",
+        title: "Failed updating profile",
+        description: "Something went wrong.",
+      });
+    },
+  });
+
+  useEffect(() => {
```

Comment:
I think we need to fetch data initially from the server component and pass the `profile` as prop. `useEffect` is redundant here.